src/coreclr/tools/aot/Mono.Linker.Tests/Extensions/CecilExtensions.cs

```diff
@@ -179,7 +179,7 @@ public static string GetDisplayName(this MethodReference method)
             if (methodDefinition != null && (methodDefinition.IsSetter || methodDefinition.IsGetter))
             {
                 // Append property name
-                string name = methodDefinition.IsSetter ? string.Concat(methodDefinition.Name.AsSpan(4), ".set") : string.Concat(methodDefinition.Name.AsSpan(4), ".get");
+                string name = methodDefinition.IsSetter ? string.Concat(methodDefinition.Name.Replace("set_",""), ".set") : string.Concat(methodDefinition.Name.Replace("get_",""), ".get");
```

Comment:
Why is this change needed?

Reply:
In the case of Explicit implementations -> `ExplicitImplementationClassWithoutRequires.Mono.Linker.Tests.Cases.RequiresCapability.RequiresAttributeMismatch.IBaseWithRequires.get_Property()`
The namespace is -> `ExplicitImplementationClassWithoutRequires`
The name is -> `Mono.Linker.Tests.Cases.RequiresCapability.RequiresAttributeMismatch.IBaseWithRequires.get_Property()`
So deleting the first 4 characters doesnt delete the `get_` deletes `Mono`

Reply:
Interesting - good catch.

It's not very robust though, if I call the interface `ILetsNotForget_One` then this will rename it to `ILetsNotFor_One` and it will probably break...
But it's probably not necessary to fix - until we hit it.

Reply:
This logic is duplicated from illink - should we fix it there too: https://github.com/dotnet/linker/blob/87539d443b197b63b99fabf603e20811bad632c5/src/linker/Linker/MethodReferenceExtensions.cs#L20

(And if we're fixing it for illink where this is part of the shipping product I think, maybe we should do this more carefully and only do this for `.get_`. The dot is rarely a user controller character. Underscore can be part of a legit property name.)

Reply:
This change got a little messy because the file had the runtime format and its a linker file. But I change the code to:
```C#
string name = methodDefinition.IsSetter ?
    string.Concat (methodDefinition.Name.StartsWith ("set_") ? methodDefinition.Name.AsSpan (4) : methodDefinition.Name.Replace (".set_", "."), ".set") :
    string.Concat (methodDefinition.Name.StartsWith ("get_") ? methodDefinition.Name.AsSpan (4) : methodDefinition.Name.Replace (".get_", "."), ".get");
```
the method definition of non Explicit name does not include the dot '.' in the name its only `set_<NameOfProperty>` while in the other case it does have the dot. I tried to take into account that by checking if the name starts with `set_/get_`

Reply:
Yup, that's what I had in mind. It could probably help to extract it to a helper method (`string GetPropertyNameFromAcessorName(string methodName)`?)